### PR TITLE
[mac] add platform macro for ACK FP

### PR DESCRIPTION
**Pull request by simonlingoogle**
The commits add a new platform macro for ACK FP:
- platforms that set ACK FP correctly should
 `#define OPENTHREAD_CONFIG_PLATFORM_SET_ACK_FP_CORRECTLY 1`
- other platforms should `#define OPENTHREAD_CONFIG_PLATFORM_SET_ACK_FP_CORRECTLY 0`

The commits check if `OPENTHREAD_CONFIG_PLATFORM_SET_ACK_FP_CORRECTLY` is set to 1. If not, it uses source match to determine ACK FP for data requests, which is highly likely to be correct. 

The direct reason for introducing this flag is because CC2538 OT failed to pass 4 TH tests cases due to redundant `empty data frame` caused by PR #4131 . PR #4131 relies on platform radio driver to set ACK FP correctly, unfortunately this is not the case for most platforms (only nrf52840, posix, and cc2538 with PR #4263 set ACK FP correctly). 

We expect this issue to impact all platforms that do not implement ACK FP properly yet. So, this PR is a workaround for this issue on other platforms. 

Platforms that latter implement ACK FP correctly should change `OPENTHREAD_CONFIG_PLATFORM_SET_ACK_FP_CORRECTLY` to 1

**This RP conflicts with PR #4263 on `examples/platforms/cc2538/radio.c`**

### CODE REVIEW

**Comment by simonlingoogle** [comment]
> I am fine with this change as as a work-around to pass the TH tests (on all platforms), but I think for longer term we should aim to add support for FP to be set correctly on all radio platforms.

Agree. @abtink 
Also, PR #4131 is not the root cause of this issue. The root cause is to be identified.

File: examples/platforms/cc1352/openthread-core-cc1352-config.h [hunk 1]
```diff
@@ -68,4 +68,12 @@
  */
 #define OPENTHREAD_CONFIG_NCP_UART_ENABLE 1
 
+/**
+ * @def OPENTHREAD_CONFIG_PLATFORM_SET_ACK_FP_CORRECTLY
```
[the remark this answers is not in the document]
Comment (simonlingoogle) [reply examples/platforms/cc1352/openthread-core-cc1352-config.h]: any suggestions ?

[the remark this answers is not in the document]
Comment (librasungirl) [reply examples/platforms/cc1352/openthread-core-cc1352-config.h]: I did think for a while when commenting previously... not good at naming either.
Maybe `OPENTHREAD_CONFIG_RET_FRAME_PENDING_BIT_CAP`?

[the remark this answers is not in the document]
Comment (simonlingoogle) [reply examples/platforms/cc1352/openthread-core-cc1352-config.h]: What about `OPENTHREAD_CONFIG_PLATFORM_RADIO_RET_FRAME_PENDING_BIT`?

[the remark this answers is not in the document]
Comment (jwhui) [reply examples/platforms/cc1352/openthread-core-cc1352-config.h]: > > I am fine with this change as as a work-around to pass the TH tests (on all platforms), but I think for longer term we should aim to add support for FP to be set correctly on all radio platforms.
> 
> Agree. @abtink
> Also, PR #4131 is not the root cause of this issue. The root cause is to be identified.

If we want to force platforms to implement this properly, is it better not to introduce this config option?

**Comment by jwhui** [comment]
> > I am fine with this change as as a work-around to pass the TH tests (on all platforms), but I think for longer term we should aim to add support for FP to be set correctly on all radio platforms.
> 
> Agree. @abtink
> Also, PR #4131 is not the root cause of this issue. The root cause is to be identified.

If we want to force platforms to implement this properly, is it better not to introduce this config option?

**Comment by abtink** [comment]
@jwhui I fully agree. I also do prefer if it is addressed in platform radio code. I am fine with adding such a work-around if it's urgent to unblock.

The radio code itself can use the same work-around, i.e. if driver itself does not provide the info about FP in the emitted ack, it can possibly use a src match table lookup associated with the dst address of the frame to determine/guess this.

**Comment by librasungirl** [comment]
If we could force platforms to implement this properly, it would be definitely a better option.

@jwhui How could we make it happen? Raise one issue to note that individual platform should resolve this to get Thread Certified?

**Comment by jwhui** [comment]
> @jwhui How could we make it happen? Raise one issue to note that individual platform should resolve this to get Thread Certified?

I think it makes sense to raise a GitHub Issue describing the problem and listing platforms that are currently affected.

**Comment by jwhui** [comment]
Submitted #4358 as discussed above.

**Pull request closed without merging**